purchase_request/models/purchase_request.py

```diff
@@ -0,0 +1,251 @@
+                               track_visibility='onchange')
+    state = fields.Selection(selection=_STATES,
+                             string='Status',
+                             index=True,
+                             track_visibility='onchange',
+                             required=True,
+                             copy=False,
+                             default='draft')
+    is_editable = fields.Boolean(string="Is editable",
+                                 compute="_compute_is_editable",
+                                 readonly=True)
+
+    picking_type_id = fields.Many2one('stock.picking.type',
+                                      'Picking Type', required=True,
+                                      default=_default_picking_type)
+
+    @api.multi
+    def copy(self, default=None):
+        default = dict(default or {})
+        self.ensure_one()
+        default.update({
+            'state': 'draft',
+            'name': self.env['ir.sequence'].get('purchase.request'),
+        })
+        return super(PurchaseRequest, self).copy(default)
+
+    @api.model
+    def create(self, vals):
+        request = super(PurchaseRequest, self).create(vals)
+        if vals.get('assigned_to'):
+            request.message_subscribe_users(user_ids=[request.assigned_to.id])
+        return request
+
+    @api.multi
+    def write(self, vals):
+        res = super(PurchaseRequest, self).write(vals)
+        for request in self:
+            if vals.get('assigned_to'):
+                self.message_subscribe_users(user_ids=[request.assigned_to.id])
+        return res
+
+    @api.multi
+    def button_draft(self):
+        for rec in self:
+            rec.state = 'draft'
+        return True
+
+    @api.multi
+    def button_to_approve(self):
+        for rec in self:
+            rec.state = 'to_approve'
+        return True
+
+    @api.multi
+    def button_approved(self):
+        for rec in self:
+            rec.state = 'approved'
+        return True
+
+    @api.multi
+    def button_rejected(self):
+        for rec in self:
+            rec.state = 'rejected'
+        return True
+
+
+class PurchaseRequestLine(models.Model):
+
+    _name = "purchase.request.line"
+    _description = "Purchase Request Line"
+    _inherit = ['mail.thread', 'ir.needaction_mixin']
+
+    @api.multi
+    @api.depends('product_id', 'name', 'product_uom_id', 'product_qty',
+                 'analytic_account_id', 'date_required', 'specifications')
+    def _compute_is_editable(self):
+        for rec in self:
+            if rec.request_id.state in ('to_approve', 'approved', 'rejected'):
+                rec.is_editable = False
+            else:
+                rec.is_editable = True
+
+    @api.multi
+    def _compute_supplier_id(self):
+        for rec in self:
+            if rec.product_id:
+                for product_supplier in rec.product_id.seller_ids:
+                    rec.supplier_id = product_supplier.name
```

Comment:
This is information for the Purchasing Department so they are able to know what is the supplier for that product.